src/lazy/expanded/compiler.rs

```diff
@@ -1412,44 +1387,62 @@ impl<'top, D: Decoder> TdlSExpKind<'top, D> {
             }
         };
 
-        let operation_name = TemplateCompiler::expect_symbol_text("operation name", operation)?;
+        // In most cases, an expression in this position is a macro ID. Try to resolve it.
+        if let Some(macro_ref) = TemplateCompiler::resolve_macro_id_expr(tdl_context, operation)? {
+            return Ok(TdlSExpKind::MacroInvocation(macro_ref, expressions));
+        }
 
+        // If look-up fails to resolve to a macro, it might be a special form.
+        Self::expect_special_form(tdl_context, operation, expressions)
+    }
+
+    fn expect_special_form(
```

Comment:
🪧 I extracted this into a helper method so the logic for special forms was more encapsulated.